os/mm/mm_heap/mm_manage_allocfail.c

```diff
@@ -90,6 +90,14 @@ void mm_manage_alloc_fail(struct mm_heap_s *heap, int startidx, int endidx, size
 {
 	irqstate_t flags = enter_critical_section();
 
+#ifdef CONFIG_SMP
+	/* If SMP is enabled then we need to pause all the other cpu's immediately
+	 * because the kernel state might be invalid at this point. If we don't pause
```

Comment:
I dont think the kernel state will be invalid. Let us just mention that we pause the other cpu to avoid mixed logs.